express-test/docker-compose.yml

```
@@ -0,0 +1,15 @@
+version: '2'
+services:
+  postgres:
+    image: postgres:16-alpine
+    environment:
+      POSTGRES_USER: express
+      POSTGRES_PASSWORD: express
+      POSTGRES_DB: express
+    ports:
+      - "5432:5432"
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready"]
```

Given that a custom user and DB is used I suggest to use those ones instead of falling back to default and checking that from `root` role. In order to do that we can pass explicitly `--username` and `--dbname`.

While here we can also avoid spawning a shell and directly invoke `pg_isready` by using `CMD` instead of `CMD-SHELL`.

Here suggested healthcheck:

```suggestion
      test: ["CMD", "pg_isready", "--username", "express", "--dbname", "express"]
```

(Without that suggestion, if you look at the logs I would expect:

```
tmp-postgres-1  | 2024-01-02 13:09:12.674 UTC [93] FATAL:  role "root" does not exist
```

...or similar and this suggestion should address them.)

---

This is pretty useful and you answered to my doubts about that error. Very appreciated, thanks.

---

No problem!